frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java

```diff
@@ -0,0 +1,95 @@
+ * create an instance of this fragment.
+ */
+public class ListingsFragment extends Fragment {
+    final static String TAG = "ListingsFragment";
+    private RecyclerView recyclerView;
+    private ArrayList<RecyclerData> recyclerDataArrayList;
+
+    // TODO: Rename parameter arguments, choose names that match
+    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+    private static final String ARG_PARAM1 = "param1";
+    private static final String ARG_PARAM2 = "param2";
+
+    // TODO: Rename and change types of parameters
+    private String mParam1;
+    private String mParam2;
+
+    public ListingsFragment() {
+        // Required empty public constructor
+    }
+
+    /**
+     * Use this factory method to create a new instance of
+     * this fragment using the provided parameters.
+     *
+     * @param param1 Parameter 1.
+     * @param param2 Parameter 2.
+     * @return A new instance of fragment ListingsFragment.
+     */
+    // TODO: Rename and change types and number of parameters
+    public static ListingsFragment newInstance(String param1, String param2) {
+        ListingsFragment fragment = new ListingsFragment();
+        Bundle args = new Bundle();
+        args.putString(ARG_PARAM1, param1);
+        args.putString(ARG_PARAM2, param2);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mParam1 = getArguments().getString(ARG_PARAM1);
+            mParam2 = getArguments().getString(ARG_PARAM2);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_listings, container, false);
+        recyclerView = view.findViewById(R.id.idCourseRV);
+
+        // created new array list..
+        recyclerDataArrayList=new ArrayList<>();
+
+        // added data to array list
+        recyclerDataArrayList.add(new RecyclerData("House", R.drawable.ic_listings_image));
+        recyclerDataArrayList.add(new RecyclerData("Casa", R.drawable.ic_listings_image));
+        recyclerDataArrayList.add(new RecyclerData("Maison", R.drawable.ic_listings_image));
+        recyclerDataArrayList.add(new RecyclerData("Haus", R.drawable.ic_listings_image));
+        recyclerDataArrayList.add(new RecyclerData("家", R.drawable.ic_listings_image));
+
+        // added data from arraylist to adapter class.
+        RecyclerViewAdapter adapter=new RecyclerViewAdapter(recyclerDataArrayList,view.getContext());
+
+        // setting grid layout manager to implement grid view.
+        // in this method '2' represents number of columns to be displayed in grid view.
+        GridLayoutManager layoutManager=new GridLayoutManager(view.getContext(),2);
```

Comment:
magic number? maybe define num_cols as 2 somewhere in the file etc.

Reply:
good call!